Update rules tree to display newly added rules inside group

pawanbhttr opened the pull request:
## Description
<!--
  Please describe the changes and add a link to the related issue(s) #
-->

## Resolves #1901 

I have updated or-rules-tree component to add newly created rules inside their respective group if group is selected when creating rules.

## Before Enhancement

![image](https://github.com/user-attachments/assets/488f5dd0-cc93-4096-8eea-45497236dcda)

## After Enhancement

https://github.com/user-attachments/assets/c9a07447-4c67-4e5c-8108-d6e1c32152e1

## Checklist
<!--
  With all these boxes checked this PR conforms to our Definition of Done.
-->

- [ ] 1. Acceptance criteria of the linked issue(s) are met
- [ ] 2. Tests are written and all tests pass
- [ ] 3. Changes are manually tested by you and the reviewer

<!-- 
  Thank you for your contribution <3 
-->

MartinaeyNL commented:
Thanks for the PR!

I'll try to find the time to test and review this.
Looking promising so far! 👍

MartinaeyNL commented on ui/component/or-rules/src/or-rule-tree.ts:
```diff
@@ -127,7 +127,11 @@ export class OrRuleTree extends OrTreeMenu {
                 this._loadRulesets(this.global).then(rulesets => this.rules = rulesets);
             } else {
-                this.nodes = [...this._getRuleNodes(this.rules), ...this._cachedEmptyGroupNodes];
+                const ruleNodes = this._getRuleNodes(this.rules);
+                const ruleNodeIds = new Set(ruleNodes.map(node => node.id));
+                const emptyGroups = this._cachedEmptyGroupNodes
+                    .filter(cachedNode => !ruleNodeIds.has(cachedNode.id));
+                this.nodes = [...ruleNodes, ...emptyGroups];
```
I was curious why this code was necessary, so I pulled your branch to my local system.
To my understanding, it's similar to line 138, where groups without any rules are filtered out.

Can you add a short comment in code, so it's clear why/what it's meant to do?

pawanbhttr commented:
Hi @MartinaeyNL,
I have updated the code as suggested and added comments to explain the functionality of the code. I am working on extending feature, will let you know after I complete it.

pawanbhttr commented:
Hi @MartinaeyNL , 
I have updated the PR to add extend feature in tree when new rules is added to the group. Please review it.
Thank You

MartinaeyNL commented on ui/component/or-rules/src/or-rule-tree.ts:
```diff
@@ -609,4 +629,23 @@ export class OrRuleTree extends OrTreeMenu {
         }
     }
+
+    /**
+     * Programmatically finds a group by its ID and sets it to be expanded.
+     * This is called by the parent component after a new rule is added to a group.
+     * @param {string} groupId The ID of the group to expand.
+     */
+    public expandGroup(groupId: string) {
+        // Wait for any pending Lit updates to complete to ensure the node list is stable.
+        this.updateComplete.then(() => {
+            const groupNode = this.nodes.find(node => node.id === groupId && node.children);
+
+            if (groupNode) {
+                groupNode.expanded = true;
+                // re-render with the change.
+                this.requestUpdate();
+            }
+        });
+    }
```
Can you move this to the base class of `or-tree-menu` ?
So the function is available for every tree menu in the UI 👍

MartinaeyNL reviewed (commented):
Thanks for addressing the changes!

I've only got one final remark. I'm happy to merge after that 💪
